lib/super_digit.rb

```diff
@@ -1,15 +1,30 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(log n), b/c #steps grows every 10th power, not linearly
+# Space Complexity - O(log n)
+### I couldn't figure out a way to incorporate dynamic programming into this...
+### would I even benefit from dynamic programming...?
 def super_digit(n)
-   
+  if n < 10
+    return n
+  else
+    array = num2array(n)
+    return super_digit(array.sum)
+  end
 end
-  
 
-# Time Complexity - ?
-# Space Complexity - ?
-def refined_super_digit(n, k)
-    
+def num2array(number)
+  # O(log n) time & space
+  # takes something like 123 and returns [1,2,3]
+  array = number.to_s.split('')
+  return array.map { |num| num.to_i}
 end
-    
\ No newline at end of file
+
+
+# Time Complexity - O(log n)
+# Space Complexity - O(log n)
+def refined_super_digit(n, k)
```

Comment:
This solution works, but you can do better by applying dynamic programming here.  

Consider, you're solving for superdigit(n) several times.  Can you apply dynamic programming here?